src/lightning_app/core/work.py

```diff
@@ -644,6 +644,19 @@ def stop(self):
         app = _LightningAppRef().get_current()
         self._backend.stop_work(app, self)
 
+    def delete(self):
```

Comment:
shall we redefine buildin `__del__`?

Reply:
I thought about it but let's wait a bit more for introducing it. This might have side effect and better to explore this a bit more. For another PR.